app/backplane/sensor_module/include/c_sensing_tenant.h

```diff
@@ -0,0 +1,23 @@
+#ifndef C_SENSING_TENANT_H
+#define C_SENSING_TENANT_H
+
+#include <f_core/os/c_tenant.h>
+
+class CSensingTenant : public CTenant {
```

Comment:
Do you think we would want to give ourselves an easy way to spin up sensing tenants and get rid of a bit of duplication. This got me thinking to have CSensingTenant inherit from CTenant and take a frequency/period,  override run to do the timing and on its inside call a  child-class-of-it-supplied Measure function which does the actual thinking. 
IE.
```
                CTenant
                   |
             CSensingTenant
          /                  \
 100hzTenant                1hzTenant
```

```cpp
class CSensingTenant : public CTenant{
      CSensingTenant(period);
      virtual void Measure() = 0;
      void Run(){
           while(going){
                 // handle timing
                 Measure();
            }
      }
};

class SensorMod100Hz : public CSensingTenant{
public:
      SensorMod100Hz() : CSensingTenant(100hz){
           // other setup stuff
     }
     void Measure(){
          // read from sensors
    }
private:
     // Devices
     // queue for data
    // anything else you need
}

```

Reply:
Interesting idea for the future

Reply:
gonna make it a isseu